bundles/org.openhab.binding.automower/src/main/java/org/openhab/binding/automower/internal/things/AutomowerHandler.java

```diff
@@ -0,0 +1,301 @@
+                logger.info("No valid polling interval specified. Using default value: {}s", automowerPollingIntervalS);
+            } else {
+                automowerPollingIntervalS = config.getPollingInterval();
+            }
+            automowerPollingJob = scheduler.scheduleWithFixedDelay(new AutomowerPollingRunnable(), 1,
+                    automowerPollingIntervalS, TimeUnit.SECONDS);
+        }
+    }
+
+    private void stopAutomowerPolling() {
+        if (automowerPollingJob != null && !automowerPollingJob.isCancelled()) {
+            automowerPollingJob.cancel(true);
+            automowerPollingJob = null;
+        }
+    }
+
+    private boolean isValidResult(Mower mower) {
+        return mower != null && mower.getAttributes() != null && mower.getAttributes().getMetadata() != null
+                && mower.getAttributes().getBattery() != null && mower.getAttributes().getSystem() != null;
+    }
+
+    private boolean isConnected(Mower mower) {
+        if (mower != null && mower.getAttributes() != null && mower.getAttributes().getMetadata() != null
+                && mower.getAttributes().getMetadata().isConnected()) {
+            return true;
+        }
+        return false;
+    }
+
+    private synchronized void updateAutomowerState() {
+        if (System.nanoTime() - lastQueryTime > maxQueryFrequencyNanos) {
+            lastQueryTime = System.nanoTime();
+            String id = automowerId.get();
+            try {
+                Mower mower = getAutomowerBridge().getAutomowerStatus(id);
+
+                if (isValidResult(mower)) {
+                    initializeProperties(mower);
+
+                    updateChannelState(mower);
+
+                    if (isConnected(mower)) {
+                        updateStatus(ThingStatus.ONLINE);
+                    } else {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                "@text/comm-error-mower-not-connected-to-cloud");
+                    }
+                } else {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            "@text/comm-error-query-mower-failed");
+                }
+            } catch (AutomowerCommunicationException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "@text/comm-error-query-mower-failed");
+                logger.warn("Unable to query automower status for:  {}", id, e);
+            }
+        }
+    }
+
+    private synchronized void sendAutomowerCommand(Command command) {
+        String id = automowerId.get();
+        try {
+            if (getAutomowerBridge().sendAutomowerCommand(id, command.toString(), commandDuration.get())) {
+                updateState(CHANNEL_COMMAND_RESPONSE,
+                        new StringType("Successfully sent command " + command.toString()));
```

Comment:
Is there a reason why you feed the result back into a Channel instead of logging it?

Reply:
So I can create an item where I can view the response to my command in the app... Basically you are right, it is for debugging mostly, because my command should have an affect on some other channels (mower action,...). If I send a command and nothing would change I am interested in that.

Reply:
It should be logged if the command could not be executed, because of a communication problem or bug. But if this is expected to fail e.g. because the mower is in a state where the command is rejected, I don't see any better solution to feedback via a Channel.

Reply:
Ok, thanks. I still removed it. As I changed the whole command sending to actions instead of channels I also dropped the feedback through the channel. Usually the command will reflect in a change of the mower state, so you should see if it works.